### PR TITLE
Fix ominous warning: jar coordinate ... already loaded

### PR DESCRIPTION
**Pull request by jakelandis**
This change to ensure that Java and JRuby agree upon the version of Jackson. This downgrades Java from 2.7.4 to 2.7.3 (which is the JRuby version).

Fixes: #6781 
-----------

Steps used to make this change:

clean checkout 
downgrade version to 2.7.3 in logstash-core/build.gradle
add in com.fasterxml.jackson.core:jackson-annotations:2.7.3 since the transtive version is a mis-match
```
+--- com.fasterxml.jackson.core:jackson-databind:2.7.3
|    +--- com.fasterxml.jackson.core:jackson-annotations:2.7.0
```
./gradlew gemspec_jars 
rake test:install-core

### CODE REVIEW

**Comment by jakelandis** [comment]
Jenkins test this please

File: logstash-core/build.gradle [hunk 1]
```diff
@@ -108,10 +108,11 @@ dependencies {
     compile 'org.apache.logging.log4j:log4j-api:2.6.2'
     compile 'org.apache.logging.log4j:log4j-core:2.6.2'
     runtime 'org.apache.logging.log4j:log4j-slf4j-impl:2.6.2'
-    compile 'com.fasterxml.jackson.core:jackson-core:2.7.4'
-    compile 'com.fasterxml.jackson.core:jackson-databind:2.7.4'
-    compile 'com.fasterxml.jackson.module:jackson-module-afterburner:2.7.4'
-    compile 'com.fasterxml.jackson.dataformat:jackson-dataformat-cbor:2.7.4'
+    compile 'com.fasterxml.jackson.core:jackson-core:2.7.3'
```
Comment (andrewvc) [inline logstash-core/build.gradle]: Can we throw a comment in here that we're pinned to this version due to JRuby?

**Comment by andrewvc** [comment]
Jenkins please test this

File: logstash-core/build.gradle (hunk 1 above)
Comment (robbavey) [inline logstash-core/build.gradle]: Is it jruby itself we are pinned to it, or is Guy's jruby wrapper for jackson (jrjackson - https://github.com/guyboertje/jrjackson)?

Comment (jakelandis) [inline logstash-core/build.gradle]: I can place a comment in here.  @robbavey - you are correct we need to stay in sync with the Java Jackson version used by Guy's JRuby wrapper gem.

**Comment by andrewvc** [comment]
@jakelandis any idea on the jenkins test failure here?

**Comment by jakelandis** [comment]
@andrewvc - Not yet.  Can't reproduce on my mac, trying now on my Linux box.

**Comment by jakelandis** [comment]
This fails only on the Jenkins build machine (works fine locally and Travis).  

I added ```export JARS_DEBUG=true``` to ci/ci_test.sh and echo the [bundler output on failure](https://github.com/elastic/logstash/pull/7614/commits/fc62b41e60401233588b27da648d80e4e2ffbaff), and discovered that the new JAR(s) can not be downloaded on the Jenkins build machine due to a ```Connection refused (Connection refused)``` error. 

I have logged a support ticket to address the Jenkins issue. Until that is fixed, this issue is blocked.

**Comment by jakelandis** [comment]
Jenkins please test this

**Comment by jakelandis** [comment]
Jenkins please test this

**Comment by jakelandis** [comment]
@andrewvc  - This should be ready for review now. 

The Jenkins tests were failing due to an invalid ~/.m2/settings.xml which was referencing a non-existent localhost proxy, resulting in not being able to talk to maven. It's fixed now by manually removing the invalid proxy settings from the build machines.

I _think_ there maybe a bug in the jar-dependencies project, but nothing obvious from a quick scan of their code...if not that project, I have no idea how that invalid ~/.m2/settings.xml ended up on the build machines.  

For posterity and search here is the invalid settings.xml 
```
<settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0
  https://maven.apache.org/xsd/settings-1.0.0.xsd">
  <proxies>
    
    <proxy>
      <id>local.dev.0</id>
      <active>true</active>
      <protocol>http</protocol>
      <host>local.dev</host>
      <port>9898</port>
      <username>a</username>
      <password>b</password>
    </proxy>
  
    <proxy>
      <id>local.dev.1</id>
      <active>true</active>
      <protocol>https</protocol>
      <host>local.dev</host>
      <port>9898</port>
      <username>c</username>
      <password>d</password>
    </proxy>
  
  </proxies>
</settings>
```

**Comment by andrewvc** [comment]
LGTM

**Pull request closed without merging**